conjure-java-core/src/main/java/com/palantir/conjure/java/types/BeanBuilderGenerator.java

```diff
@@ -106,6 +106,10 @@ private TypeSpec generate(ObjectDefinition typeDef) {
                 .addAnnotation(
                         AnnotationSpec.builder(JsonIgnoreProperties.class)
                                 .addMember("ignoreUnknown", "$L", true)
+                                .build())
+                .addAnnotation(
+                        AnnotationSpec.builder(SuppressWarnings.class)
+                                .addMember("value", "$S", "unchecked")
```

Comment:
At a minimum I think we need a comment here justifying why we add this to _all_ builders... I just ran it locally and got:

```
> Task :conjure-java-core:compileIntegrationInputJava FAILED
/Users/dfox/conjure-java/conjure-java-core/src/integrationInput/java/com/palantir/product/CovariantOptionalExample.java:107: warning: [unchecked] unchecked cast
            this.item = (Optional<Object>) Preconditions.checkNotNull(item, "item cannot be null");
                                                                     ^
  required: Optional<Object>
  found:    Optional<CAP#1>
  where CAP#1 is a fresh type-variable:
    CAP#1 extends Object from capture of ?
error: warnings found and -Werror specified
1 error
1 warning
```